src/app/fdctl/keygen.c

```diff
@@ -64,7 +83,24 @@ generate_keypair( const char * keyfile,
 void
 keygen_cmd_fn( args_t *         args,
                config_t * const config ) {
-  (void)args;
 
-  generate_keypair( config->consensus.identity_path, config );
+  if( FD_LIKELY( args->keygen.key == KEYGEN_KEY_IDENTITY ) ) {
+    if( FD_UNLIKELY( !strcmp( config->consensus.identity_path, "" ) ) ) {
+      snprintf1( config->consensus.identity_path,
+              sizeof(config->scratch_directory) + 15,
+              "%s/identity.json",
+              config->scratch_directory );
+    }
+    generate_keypair( config->consensus.identity_path, config );
+  }
+
+  if( FD_LIKELY( args->keygen.key == KEYGEN_KEY_VOTE_ACCOUNT ) ) {
+    if( FD_LIKELY( !strcmp( config->consensus.vote_account_path, "" ) ) ) {
+      snprintf1( config->consensus.vote_account_path,
```

Comment:
Print an error here instead of default initializing this. Something like

FD_LOG_ERR(( "Cannot create a vote account keypair because your validator is not configured to vote. Please set [consensus.vote_account_path] in your configuration file." ));